scipy/stats/_continuous_distns.py

```diff
@@ -10481,41 +10502,51 @@ def _shape_info(self):
         return [ib, ic]
 
     def _argcheck(self, b, c):
-        b = float(b)  # ensure float
         return (b != 0.) & (c > 1.)
 
     def _get_support(self, b, c):
         return self.a, c
 
     def _pdf(self, x, b, c):
+        b = np.asarray(b, dtype=x.dtype)  # avoid int to negative int power
         return b * x**-(b+1) / (1 - 1/c**b)
 
     def _logpdf(self, x, b, c):
         return np.log(b) - np.log(-np.expm1(-b*np.log(c))) - (b+1)*np.log(x)
```

Comment:
The first two $b$s here come from $\log \left( \frac{b}{1-c^{-b}} \right$, so the signs cancel one another out. They can be $|b|$.